OrbitGl/App.cpp

```diff
@@ -120,35 +166,47 @@ void OrbitApp::OnKeyAndString(uint64_t key, std::string str) {
 }
 
 void OrbitApp::OnCallstack(CallStack callstack) {
-  Capture::GSamplingProfiler->AddUniqueCallStack(callstack);
+  main_thread_executor_->Schedule([callstack = std::move(callstack)] {
```

Comment:
I have to agree that this seems like *a lot* of scheduling actions if a new one is scheduled for each event, even if this is not done for `Timer`s which are the most frequent ones. Not sure how a reasonably quick fix would look like though. But as this is not directly related to the issue with `CaptureClient` (right?), was this actually causing problems?

Reply:
This should at least be measured to understand the performance impact, o/w it seems to be hard to make a decision here.

Reply:
I have not seen crashes because of this - but this is obviously not thread_safe so I think we need to fix it.

Reply:
ok, I made sampling profiler thread-safe, lets try again.